packages/block-library/src/navigation/style.scss

```diff
@@ -1,236 +1,269 @@
-.wp-block-navigation {
-	& > ul {
-		display: block;
+$navigation-min-height: 60px;
+
+/*
+* Frontend: reset the default list styles
+*/
+
+.wp-block-navigation > ul {
+	display: block;
+	list-style: none;
+	margin: 0;
+	padding-left: 0;
+
+	@include break-small {
+		display: flex;
+		flex-wrap: wrap;
+	}
+
+	// Submenu
+	ul {
 		list-style: none;
-		margin: 0;
-		max-width: none;
 		padding-left: 0;
-		position: relative;
+		margin-top: 0;
+		margin-left: 0;
 
-		@include break-small {
-			display: flex;
-			flex-wrap: wrap;
+		li {
+			margin: 0;
 		}
+	}
+}
 
-		ul {
-			padding-left: 0;
-		}
+/*
+* Frontend: styles for submenu flyout
+*/
 
-		li {
-			position: relative;
-			z-index: 1;
+.wp-block-navigation > ul {
+	li {
+		z-index: 1;
 
-			&:hover,
-			&:focus-within {
-				cursor: pointer;
-				z-index: 99999;
-			}
+		&:hover,
+		&:focus-within {
+			cursor: pointer;
+			z-index: 99999;
+		}
 
-			// Submenu Display
-			&:hover > ul,
-			&:focus-within > ul,
-			& ul:hover,
-			& ul:focus {
-				visibility: visible;
-				opacity: 1;
-				display: block;
-			}
+		// Submenu Display
+		&:hover > ul,
+		&:focus-within > ul,
+		& ul:hover,
+		& ul:focus {
+			visibility: visible;
+			opacity: 1;
+			display: flex;
+			flex-direction: column;
 		}
+	}
 
-		// ToDo: move navigation-link styles to navigation-link/style.scss
-		& > li {
+	& > li ul {
+		position: absolute;
+		left: 0;
+		top: 80%;
+		min-width: max-content;
+		opacity: 0;
+		transition: opacity 0.15s linear, transform 0.15s linear, right 0s 0.15s;
+		transform: translateY(0.6rem);
+		visibility: hidden;
+	}
+}
 
-			& > a {
-				display: flex;
-				align-items: center;
-				padding-left: 0;
+/*
+* Styles shared between editor and frontend
+*/
+.wp-block-navigation,
+.wp-block-navigation .block-editor-block-list__layout {
+	display: flex;
+	flex-wrap: wrap;
+}
 
-				@include break-small {
-					padding-left: 16px;
-				}
-			}
+.wp-block-navigation {
 
-			&:first-of-type > a {
-				padding-left: 0;
+	// Min height and vertical alignment (main menu - level 0)
+	&,
+	> .wp-block-navigation__container {
+		min-height: $navigation-min-height;
+		align-items: center;
+		width: 100%;
+
+		> .wp-block-navigation-link {
+			display: flex;
+			min-height: $navigation-min-height;
+			margin-top: 0;
+			margin-bottom: 0;
+		}
+	}
+
+	// Main menu
+	.wp-block-navigation-link {
```

Comment:
Should this be in the navigation-link component?

Reply:
We've been talking about where we should put this code. Although it clearly belongs to the navigation-link component, splitting out the code of the sub-menus structure in two files will make understanding the code otherwise tough labor. 
So we agreed to keep it in the same file, organized by sections. We also considered creating a new file only to implement the menu structure, but it's unneeded at least for now.